Microsoft.Toolkit.Uwp.UI.Controls.Graph/PlannerTaskList/PlannerTaskList.cs

```diff
@@ -0,0 +1,346 @@
+                {
+                    InternalPlanId = PlanId;
+                }
+            }
+            catch (Exception exception)
+            {
+                MessageDialog messageDialog = new MessageDialog(exception.Message);
+                await messageDialog.ShowAsync();
+            }
+        }
+
+        private async Task InitPlanAsync()
+        {
+            try
+            {
+                ClearTasks();
+                MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
+                await graphService.TryLoginAsync();
+                GraphServiceClient graphClient = graphService.GraphProvider;
+                IPlannerPlanBucketsCollectionPage buckets = await graphClient.Planner.Plans[PlanId].Buckets.Request().GetAsync();
+                List<PlannerBucket> bucketList = new List<PlannerBucket>();
+                while (true)
+                {
+                    foreach (PlannerBucket bucket in buckets)
+                    {
+                        bucketList.Add(bucket);
+                    }
+
+                    if (buckets.NextPageRequest == null)
+                    {
+                        break;
+                    }
+
+                    buckets = await buckets.NextPageRequest.GetAsync();
+                }
+
+                TaskFilterSource.Clear();
+                Buckets.Clear();
+                TaskFilterSource.Add(new PlannerBucket { Id = TaskTypeAllTasksId, Name = AllTasksLabel });
+                foreach (PlannerBucket bucket in bucketList)
+                {
+                    Buckets.Add(bucket);
+                    TaskFilterSource.Add(bucket);
+                }
+
+                TaskFilterSource.Add(new PlannerBucket { Id = TaskTypeClosedTasksId, Name = ClosedTasksLabel });
+                TaskType = TaskTypeAllTasksId;
+                await LoadAllTasksAsync();
+            }
+            catch (Exception exception)
+            {
+                MessageDialog messageDialog = new MessageDialog(exception.Message);
+                await messageDialog.ShowAsync();
+            }
+        }
+
+        private void ClearTasks()
+        {
+            if (_allTasks.Count > 0)
+            {
+                foreach (var task in _allTasks)
+                {
+                    task.PropertyChanged -= TaskViewModel_PropertyChanged;
+                }
+
+                _allTasks.Clear();
+            }
+
+            if (Tasks.Count > 0)
+            {
+                Tasks.Clear();
+            }
+        }
+
+        private async Task LoadAllTasksAsync()
+        {
+            try
+            {
+                MicrosoftGraphService graphService = MicrosoftGraphService.Instance;
+                await graphService.TryLoginAsync();
+                GraphServiceClient graphClient = graphService.GraphProvider;
+                IPlannerPlanTasksCollectionPage tasks = await graphClient.Planner.Plans[PlanId].Tasks.Request().GetAsync();
```

Comment:
Are we guaranteed that `graphClient.Planner.Plans[PlanId]` will never be null?

Reply:
@skendrot Yes, the `Plans` property isn't a local dictionary, this expression is to build a Graph query for remote api call, we wouldn't know if a plan exists or not until the method `GetAsync()` is called.